cli/flox-rust-sdk/src/models/pkgdb.rs

```diff
@@ -62,20 +64,44 @@ pub enum CallPkgDbError {
     ParseJSON(#[source] serde_json::Error),
     #[error("call to pkgdb failed")]
     PkgDbCall(#[source] std::io::Error),
+    #[error("couldn't get pkgdb stdout")]
+    PkgDbStdout,
+    #[error("couldn't get pkgdb stderr")]
+    PkgDbStderr,
 }
 
 /// Call pkgdb and try to parse JSON or error JSON.
 ///
 /// Error JSON is parsed into a [CallPkgDbError::PkgDbError].
 pub fn call_pkgdb(mut pkgdb_cmd: Command) -> Result<Value, CallPkgDbError> {
-    let output = pkgdb_cmd.output().map_err(CallPkgDbError::PkgDbCall)?;
+    let mut proc = pkgdb_cmd
+        .stderr(Stdio::piped())
+        .stdout(Stdio::piped())
+        .spawn()
+        .map_err(CallPkgDbError::PkgDbCall)?;
+    let stderr = proc.stderr.take();
+    let Some(stderr) = stderr else {
+        proc.kill().map_err(CallPkgDbError::PkgDbCall)?;
+        return Err(CallPkgDbError::PkgDbStderr);
+    };
```

Comment:
It will be `Some` the *first* time someone `take`s the file descriptor. `call_pkgdb` is handed a `Command` which may have arbitrary things done to it before it receives it. That's unlikely, but I'd rather that bug cause an error instead of a panic. That's why I did this.

Reply:
You do emplace an stdout and err into the "handed in" `Command`, so you control waht is `take`n. I dont see why there would be a `None` value in that case.

Reply:
That's a good point, I can't read